### PR TITLE
db: support range keys in external files

### PR DESCRIPTION
**Pull request by RaduBerinde**
#### keyspan: modernize the LevelIter test

This change improves the data-driven LevelIter test and simplfies the
code. Instead of specifying point keys, we allow extending either
point key or range key bounds directly.

#### db: support range keys in external files

We now support range keys in external files, and we support synthetic
prefix for range keys. We don't yet support synthetic suffix.

Informs #3287

### CODE REVIEW

**Review by jbowens — APPROVED** [review]
:lgtm:

> Reviewed 3 of 3 files at r1, 1 of 1 files at r2, 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/pebble/3737#-:-O1SxsYEEI0jdnThMMkm:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @itsbilal)

File: testdata/ingest_external [hunk 1]
```diff
@@ -996,3 +996,54 @@ p1_a@1: (va, .)
...
+----
+a@1: (va, .)
+b: (., [b-"c1") @2=vrange UPDATED)
+b@1: (vb, [b-"c1") @2=vrange)
+c@1: (vc, [b-"c1") @2=vrange)
+d@1: (vc, . UPDATED)
+.
+
+ingest-external
+points-and-ranges.sst synthetic-prefix=x_ bounds=(x_a,x_d) has-range-keys
```
Comment (itsbilal) [inline testdata/ingest_external:1033]: Let's try adding a case where the last key is the rangedel's end key, and yet we're at a loose bound? So something like `[a, c2)` for the vsst bounds.

**Review by itsbilal — APPROVED** [review]
LGTM

**Review by RaduBerinde — COMMENTED** [review]
TFTRs!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/pebble/3737#-:-O1TH6C-DgcQ25TfdxFk:bz5mair)* status: all files reviewed, 1 unresolved discussion (waiting on @itsbilal)

___
*[`testdata/ingest_external` line 1033 at r3](https://reviewable.io/reviews/cockroachdb/pebble/3737#gh-1672794052:-O1THoj3DmqPUY4HAqTX:b-896fix) ([raw file](https://github.com/cockroachdb/pebble/blob/74eab178f98ace19b5d994351a35e518df47fa0e/testdata/ingest_external#L1033)):*
<details><summary><i>Previously, itsbilal (Bilal Akhtar) wrote…</i></summary><blockquote>

Let's try adding a case where the last key is the rangedel's end key, and yet we're at a loose bound? So something like `[a, c2)` for the vsst bounds.
</blockquote></details>

Done.

**Comment by RaduBerinde** [comment]
TFTRs!

**Pull request merged**